File: tests/journal/error-injection.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis
# california institute of technology
# (c) 1998-2012 all rights reserved
#


"""
Verify that {error.log} works as expected
"""


def test():
    # access the package
    import journal
    # build a error channel
    error = journal.error("journal.test1")
    # deactivate it
    error.active = False

    # and make it say something
    error.log("hello world!")

    # all done
    return


# main
if __name__ == "__main__":
    test()


# end of file 

File: tests/libpyre/journal/error-injection.cc
// -*- coding: utf-8 -*-
//
// michael a.g. aïvázis
// orthologue
// (c) 1998-2015 all rights reserved
//


// for the build system
#include <portinfo>

// packages
#include <cassert>
// access to the journal header file
#include <pyre/journal.h>

// main program
int main() {

    // instantiate a error channel
    pyre::journal::error_t error("pyre.journal.test");
    error.deactivate();

    // inject all the standard manipulators and built in types
    error
        << pyre::journal::at(__HERE__)
        << pyre::journal::set("key", "value")
        << "Hello world!" << pyre::journal::newline
        << 0 << pyre::journal::newline
        << 0.0 << pyre::journal::endl;

    error
        << pyre::journal::at(__HERE__)
        << (void *)&error << pyre::journal::newline
        << std::string("Hello world!")
        << pyre::journal::endl;

    // all done
    return 0;
}

// end of file
